### ISSUE TITLE
Cygwin: rsync can't copy the jar file to host

### ISSUE THREAD

**Issue by david206** [SashaOv/oneclick-hadoop-launcher#2]
I am using Cygwin.

The error I get while running `mvn hadoop-launcher:run`:

``` shell
! The source and destination cannot both be remote.
! rsync error: syntax or usage error (code 1) at /home/lapo/package/rsync-3.0.9-1/src/rsync-3.0.9/main.c(1148) [Receiver=3.0.9]
! The source and destination cannot both be remote.
```

Added some debug logs, and found that the plugin tries to copy the jar file from `D:\My Documents\GitHub\oneclick-hadoop-launcher\hadoop-word-count\target\hadoop-word-count-1.0.jar`:

```
rsync -azv -e ssh -o D:\My Documents\GitHub\oneclick-hadoop-launcher\hadoop-word-count\target\hadoop-word-count-1.0.jar host:~/hadoop-word-count/lib/
```

The absolutePath of the source jar doesn't fit the cygwin path format.
It should be:

```
rsync -azv -e ssh -o /cygdrive/d/My\ Documents/GitHub/oneclick-hadoop-launcher/hadoop-word-count/target/hadoop-word-count-1.0.jar host:~/hadoop-word-count/lib/
```

**Comment by vitalyg** [comment]
I think the best solution for this problem is to add a cygwin mode to the plugin.
It probably should be an optional param that changes the path strings from one mode to another.

@SashaOv Do you agree?

@david206 Could you submit a pull request that does that?

**Comment by SashaOv** [comment]
I was thinking about adding a parameter to customize the rsync command so you can replace rsync by "cygwin"ed rsync that could munge the file paths before passing it on to the real rsync. 

Something like:

```
  <configuration>
     <rsync-command>my-rsync.cmd ... </rsync-command>
 </configuration>
```

Would this work for you guys?

**Comment by david206** [comment]
There is a cygwin util,  [cygpath](http://cygwin-lite.sourceforge.net/html/cygpath.html)  that converts Windows native filenames to Cygwin POSIX-style pathnames.

So probably the "cygwin"ed rsync should do:

``` bash
rsync -azv -e ssh "`cygpath -u D:\My Documents\GitHub\oneclick-hadoop-launcher\hadoop-word-count\target\hadoop-word-count-1.0.jar`" host:~/hadoop-word-count/lib/
```

**Comment by SashaOv** [comment]
David, please take a look at issue #5 . For rsync, you would configure:

```
 <commands>
    <rsync.command>
       cygwin-rsync
    </rsync.command>
</commands>
```

Where cygwin-rsync is your script that wraps the cygwin in the way you suggest. I have checked the code and the snapshot, tested locally but couldn't test is on the cluster yet: I have a problem with the VPN.

**Comment by SashaOv** [comment]
The suggested workaround has been released in v0.4. Looking forward to your feedback.
